src/lib.rs

```diff
@@ -220,49 +209,47 @@ fn resolve_project_name(args: &Args) -> Result<ProjectName> {
     }
 }
 
-fn resolve_template_dir(template_base_dir: &TempDir, args: &Args) -> Result<PathBuf> {
-    match &args.subfolder {
-        Some(subfolder) => {
-            let template_base_dir = fs::canonicalize(template_base_dir.path())?;
-            let template_dir =
-                fs::canonicalize(template_base_dir.join(subfolder)).with_context(|| {
-                    format!(
-                        "not able to find subfolder '{}' in repository {}",
-                        subfolder,
-                        args.git
-                            .as_ref()
-                            .map(|v| v.to_owned())
-                            .unwrap_or_else(|| args.path.as_ref().unwrap().display().to_string()),
-                    )
-                })?;
-
-            if !template_dir.starts_with(&template_base_dir) {
-                return Err(anyhow!(
-                    "{} {} {}",
-                    emoji::ERROR,
-                    style("Subfolder Error:").bold().red(),
-                    style("Invalid subfolder. Must be part of the template folder structure.")
-                        .bold()
-                        .red(),
-                ));
-            }
-            if !template_dir.is_dir() {
-                return Err(anyhow!(
-                    "{} {} {}",
-                    emoji::ERROR,
-                    style("Subfolder Error:").bold().red(),
-                    style("The specified subfolder must be a valid folder.")
-                        .bold()
-                        .red(),
-                ));
-            }
+fn resolve_template_dir(template_base_dir: &TempDir, subfolder: Option<&str>) -> Result<PathBuf> {
+    if let Some(subfolder) = subfolder {
+        let template_base_dir = fs::canonicalize(template_base_dir.path())?;
+        let template_dir =
+            fs::canonicalize(template_base_dir.join(subfolder)).with_context(|| {
+                format!(
+                    "not able to find subfolder '{}' in source template",
+                    subfolder
+                )
+            })?;
+
+        //TODO: Add comments to understand this check
+        // is this preventing when subfolder contains for example `../../name`?
+        if !template_dir.starts_with(&template_base_dir) {
```

Comment:
Do I understand this correctly? That it prevent from `subfolder arg` being `../../../some-name`?

Reply:
yes, that is a sandbox check mechanism, so that the repository cannot be escaped